src/nodejs/jshttpclient.nim

```diff
@@ -1,5 +1,212 @@
+  return await client.request(request)
+
+proc get*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "GET")
+  return await client.request(request)
+
+proc getContent*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "GET")
+  return await client.request(request)
+
+proc delete*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "DELETE")
+  return await client.request(request)
+
+proc deleteContent*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string; body: cstring = ""): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "DELETE", body = body)
+  return await client.request(request)
+
+proc post*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "POST")
+  return await client.request(request)
+
+proc postContent*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string; body: cstring = ""): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "POST", body = body)
+  return await client.request(request)
+
+proc put*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "PUT")
+  return await client.request(request)
+
+proc putContent*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string; body: cstring = ""): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "PUT", body = body)
+  return await client.request(request)
+
+proc patch*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "PATCH")
+  return await client.request(request)
+
+proc patchContent*(client: JsHttpClient | JsAsyncHttpClient; url: Uri | string; body: cstring = ""): Future[JsResponse] {.multisync.} =
+  let request = newJsRequest(url = cstring($url), `method` = "PATCH", body = body)
+  return await client.request(request)
+
+when isMainModule:
+  # Use with nimhttpd, see https://github.com/juancarlospaco/nodejs/issues/5
+  import std/jsconsole
+
+  let
+    client: JsHttpClient = newJsHttpClient()
+    content: cstring = client.getContent("http://localhost:1337")
+
+  console.log content
+
+runnableExamples("-r:off"):
+  from std/uri import parseUri
+
+  let client = newJsHttpClient()
+  const data = """{"key": "value"}"""
+
+  block:
+    let
+      url = parseUri("https://google.com")
+      content = client.getContent(url)
```

Comment:
```suggestion
      url: Uri = parseUri("https://google.com")
      content: cstring = client.getContent(url)
```

Add the types to the variables, I know Nim has type inference, 
but this way people reading the documentation understand the types of everything.